### ISSUE TITLE
Editing a model with `custom root type` in django admin fails with `AttributeError: 'str' object has no attribute 'root'`

### ISSUE THREAD

**Issue by noxan** [surenkov/django-pydantic-field#50]
When combining a custom root type in pydantic and making the schema optional (with null and blank flag in django model) the django admin edit and create views fail.

`AttributeError: 'str' object has no attribute 'root'`

I assume it's due the prefilled value by django admin being `None` and it someone gets turned into a `"null"` string which then fails during validation as it is no valid json.

Here the core model and schema definitions (inspired by the [pydantic example](https://docs.pydantic.dev/dev/concepts/models/#rootmodel-and-custom-root-types)):
```python
class Pets(RootModel):
    root: List[str]

    def __iter__(self):
        return iter(self.root)

    def __getitem__(self, item):
        return self.root[item]


class PetsModel(models.Model):
    pets: Optional[Pets] = SchemaField(blank=True, null=True)
```

I prepared an example to reproduce the issue: https://github.com/noxan/sample-django-pydantic-field/tree/main

**Comment by noxan** [comment]
My initial workaround was trying to rely on a default value instead, but I ended up with `AttributeError: 'list' object has no attribute 'root'` already while trying to run `makemigrations.

See https://github.com/noxan/sample-django-pydantic-field/compare/fix-with-default for the attempted fix.

**Comment by surenkov** [comment]
Hi @noxan, thank you for reporting this!

I was able to reproduce this, apparently this is a thing arising from `TypeAdapter` internal handling of unprepared `default` in `dump_python` method.

Will try to fix that on the weekend.

**Comment by noxan** [comment]
@surenkov thanks for the swift response and the pull request, highly appreciated :)

I tested it in our project and it does fix the issue partly, editing works (yeah!), but the admin create view does still fail. I could trace the issue to https://github.com/surenkov/django-pydantic-field/blob/master/django_pydantic_field/v2/forms.py#L85. When I log the values I get `initial=None` and `data="null"`.

I applied a quickfix locally with
```
if initial is None and (data == 'null' or data is None or data == ''):
    return False
```
But I assume we should rather fix this somewhere earlier to catch it earlier or prevent different default values in the first place. I don't know the codebase well yet sadly, hope this helps :)

**Comment by noxan** [comment]
Also had the issue now when I tried to save a model which already had a valid JSON object in the admin. It did not work when I changed any other field in parallel, but only saving the JSON (pydantic schema) field worked. I'll try to gather more data on this, can't reproduce it fully yet.

**Comment by surenkov** [comment]
@noxan I have updated a form field handling a bit to be more consistent with default JSONField behaviour and not fail on `has_changed` method. Checked it in conjunction with RootModel -- both create and update actions are working (either with `null=True` and passed `default=` values).

Published it as 0.3.4 on PyPI, feel free to reopen this if the issue still appears in your case.

**Comment by noxan** [comment]
@surenkov amazing, works, problem solved. thanks so much for your help and work ❤️
